.travis.yml

```diff
@@ -1,13 +1,23 @@
 sudo: required
 services: docker
-script: docker build . -t cs50/baseimage:${TRAVIS_BRANCH/master/latest}
+script: |
+  set -eo pipefail
+  [[ "$TRAVIS_BRANCH" == "master" ]] && TAG="latest" || TAG="$TRAVIS_BRANCH"
+  IMAGE="$TRAVIS_REPO_SLUG:$TAG"
+  docker build . -t "$IMAGE"
 before_deploy: docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
 deploy:
   provider: script
-  script: docker push cs50/baseimage:${TRAVIS_BRANCH/master/latest}
+  script: docker push "$IMAGE"
   on:
     all_branches: true
-    condition: $TRAVIS_BRANCH =~ ^master|python3.8$
+after_deploy: |
+  TRIGGER_TRAVIS_BUILD="/tmp/trigger-travis-build"
+  curl -fsL https://cs50.ly/trigger-travis-build -o "$TRIGGER_TRAVIS_BUILD"
+  chmod +x "$TRIGGER_TRAVIS_BUILD"
+  "$TRIGGER_TRAVIS_BUILD" -bi "$IMAGE" --tag "$TRAVIS_BRANCH" -o cs50 -r cli -v
+  # TODO figure out versioning
```

Comment:
Should we push versioned images only when we create a "release" perhaps?

Reply:
We could deploy on tag push instead. Would just be an extra manual step for us to deploy. I've actually decided to go with the tag approach for cs50/ide images because that way I know which commit corresponds to which version.

Reply:
That sounds okay, since then we can decide if/when we want to tag a version?